### ISSUE TITLE
NullReferenceException in AssertAnalyzer

### ISSUE THREAD

**Issue by rossdonald** [meziantou/Meziantou.FluentAssertionsAnalyzers#11]
Thanks for this project, I've converted all our unit test projects to use FluentAssertions. I've left the analyzers in the projects to prevent Assert being used. 

There are multiple unit test projects with the analyzer enabled but only one of them gives this error.

I tried creating a VSIX project and debugging the analyzer but I couldn't getting it going.

Any ideas why this null reference happens? or how to debug it locally?
```
##[warning]CSC(0,0): Warning AD0001: Analyzer 'Meziantou.FluentAssertionsAnalyzers.AssertAnalyzer' threw 
an exception of type 'System.NullReferenceException' with message 'Object reference not set to an instance of an object.'.
CSC : warning AD0001: Analyzer 'Meziantou.FluentAssertionsAnalyzers.AssertAnalyzer' threw 
an exception of type 'System.NullReferenceException' with message 'Object reference not set to an instance of an object.'. [D:\a\1\s\OurApp\OurApp.Web.UnitTest\OurApp.Web.UnitTest.csproj]
```
Unit test project is .NET 4.8 and some referenced projects are .NET standard 2.0

**Comment by meziantou** [comment]
In Visual Studio, you may need to expand the error in the Error Window to get the full call stack. This would help diagnose the issue.

The best way to debug this issue is to create a minimal repro. You can delete chunks of your code until the error disappears. Then, you can create a unit test with the remaining code. Finally, you can fix the code :)

> `Warning AD0001: Analyzer 'Meziantou.FluentAssertionsAnalyzers.AssertAnalyzer' threw ...`

There is not a lot of code in this class, so it should not be hard to figure out where a NRE can be thrown. Are you using NUnit, MSTests, or xUnit?

**Comment by meziantou** [comment]
I've made a quick fix to handle potential null values https://github.com/meziantou/Meziantou.FluentAssertionsAnalyzers/commit/7344748d1954bdd7faf42bd037b07dfe41745f62. Let me know if you still reproduce the exception using this new version

**Comment by rossdonald** [comment]
Thanks for the changes but it still occured. I narrowed it down to having a `throw;` statement without an exception variable like this:

```
try
{
	// code
}
catch (Exception ex)
{
	throw;
}
```

It seems to be this line that is causing the error

https://github.com/meziantou/Meziantou.FluentAssertionsAnalyzers/blob/ddcd20e456ff355eb5f56813d420d8b4457c1f97/Meziantou.FluentAssertionsAnalyzers/AssertAnalyzer.cs#L108

```
// warning AD0001: Analyzer 'Meziantou.FluentAssertionsAnalyzers.AssertAnalyzer' threw an exception of type 'System.NullReferenceException' with message 'Object reference not set to an instance of an object.'.
Exception occurred with following context:
Compilation: TestProject
IOperation: Throw
SyntaxTree: Test1.cs
SyntaxNode: throw; [ThrowStatementSyntax]@[278..284) (18,12)-(18,18)

System.NullReferenceException: Object reference not set to an instance of an object.
   at Meziantou.FluentAssertionsAnalyzers.AssertAnalyzer.AnalyzeNunitThrow(OperationAnalysisContext context) in C:\repos\Meziantou.FluentAssertionsAnalyzers\Meziantou.FluentAssertionsAnalyzers\AssertAnalyzer.cs:line 108
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerExecutor.<>c.<ExecuteOperationAction>b__63_0(ValueTuple`2 data)
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerExecutor.ExecuteAndCatchIfThrows_NoLock[TArg](DiagnosticAnalyzer analyzer, Action`1 analyze, TArg argument, Nullable`1 info)
```
I guess the `Type` property is null for this kind of statement?

**Comment by meziantou** [comment]
Thanks for the investigation and the full call stack! It should be fixed by https://github.com/meziantou/Meziantou.FluentAssertionsAnalyzers/commit/11897ae7a032b769a1a8b3b41db66aa0401b3167 and available in the version `1.0.10`

> I guess the Type property is null for this kind of statement?

`IThrowOperation.Exception` can be null in this case
